src/core/server/elasticsearch/elasticsearch_config.ts

```diff
@@ -39,7 +40,23 @@ export const config = {
       defaultValue: 'http://localhost:9200',
     }),
     preserveHost: schema.boolean({ defaultValue: true }),
-    username: schema.maybe(schema.string()),
+    username: schema.conditional(
+      schema.contextRef('dev'),
+      true,
+      schema.maybe(
```

Comment:
__optional nit:__ we can probably use `schema.maybe(schema.conditional(...))` and don't use it for the actual schemas since both of them use it anyway and it also makes it clearer that the field is optional irrespective to the mode Kibana is running in.

Reply:
Good call, that looks cleaner!